client/site-profiler/hooks/use-define-conversion-action.ts

```diff
@@ -15,41 +20,60 @@ export type CONVERSION_ACTION =
 
 export default function useDefineConversionAction(
 	domain: string,
-	whois?: WhoIs,
-	isDomainAvailable?: boolean,
-	isEligibleGoogleTransfer?: boolean,
-	hostingProvider?: HostingProvider,
-	isWordPressPlatform?: boolean
+	siteProfilerData?: DomainAnalyzerQueryResponse,
+	hostingProviderData?: HostingProviderQueryResponse,
+	urlData?: UrlData
 ): CONVERSION_ACTION | undefined {
+	const [ conversionAction, setConversionAction ] = useState< CONVERSION_ACTION | undefined >();
+
+	const isDomainAvailable = siteProfilerData?.is_domain_available;
+	const isEligibleGoogleTransfer = siteProfilerData?.eligible_google_transfer;
+
 	const isWpDomain = domain.toLowerCase().includes( 'wordpress.com' );
 	const isWpAtomicDomain = domain.toLowerCase().includes( 'wpcomstaging.com' );
-	const isA8cRegistrar = normalizeWhoisField( whois?.registrar )
+	const isWpPlatform = urlData?.platform === 'wordpress';
+
+	const isA8cRegistrar = normalizeWhoisField( siteProfilerData?.whois?.registrar )
 		.toLowerCase()
 		.includes( 'automattic' );
 	const isA8cDomain = isA8cRegistrar || isWpDomain || isWpAtomicDomain;
-	const isA8cHosting = hostingProvider?.slug === 'automattic';
+	const isA8cHosting = hostingProviderData?.hosting_provider?.slug === 'automattic';
 
-	if ( isDomainAvailable ) {
-		return 'register-domain';
-	} else if ( isA8cDomain && ! isA8cHosting ) {
-		if ( isWordPressPlatform ) {
-			return 'transfer-hosting-wp';
-		}
-		return 'transfer-hosting';
-	} else if ( ! isA8cDomain && isEligibleGoogleTransfer && isA8cHosting ) {
-		return 'transfer-google-domain';
-	} else if ( ! isA8cDomain && isEligibleGoogleTransfer && ! isA8cHosting ) {
-		if ( isWordPressPlatform ) {
-			return 'transfer-google-domain-hosting-wp';
+	useEffect( () => {
+		! domain && setConversionAction( undefined );
+	}, [ domain ] );
+
+	useEffect( () => {
+		if (
+			siteProfilerData !== undefined ||
+			hostingProviderData !== undefined ||
+			urlData !== undefined
```

Comment:
I'm getting wrong conversionAction now, the check seems weird to me.

Reply:
The idea is not to check the conversion action until the data is ready.
Before this, we get conversion action changes between different data fetching states. Now, we are waiting for data and then define the action.